tools/src/merger/OpenApiMerger.ts

```diff
@@ -134,6 +137,58 @@ export default class OpenApiMerger {
     })
   }
 
+  #fix_refs(obj: any = this._spec.components): void {
+    if (obj?.$ref !== undefined) {
+      if (obj?.description !== undefined) {
+        delete obj?.description
+      }
+    }
+
+    for (const key in obj) {
+      var item = obj[key]
+      if (_.isObject(item) || _.isArray(item)) {
+        this.#fix_refs(item)
+      }
+    }
+  }
+
+  #normalize_key(key: string): string {
+    return key
+      .replaceAll('::', '___')
+      .replaceAll('@', '__')
+      .replaceAll(':', '_')
+  }
```

Comment:
I think it's time for a quick trip down the memory lane to learn why `::`, `@`, and `:` were chosen as separators for parameter, response, and schema names. There were 2 criteria:
 1.  The character(s) used as a separator cannot appear anywhere else in the $ref. (This makes it easier to tell different component of a name apart, and easier for CodeGen to parse)
 2. The separators for parameters, responses, and schemas must be easily distinguishable from one another. (This encoding of component type into the separators helps the reader quickly identify where they are in the giant YAML file, and easier to grep for a component definition.

We can ignore criterion No.2 but I think No.1 is worth upholding. So, `.replaceAll(':', '_')` is not good. We can use `___` as the separator for all 3 types of `#components` if we let go of criteria No.2 OR come up with a different set of separators that still conform to strict OpenAPI 3.1's naming rules. Say `___`, `---`, and `...`.

Reply:
I believe we're only parsing `::` and `@` and `:` is pure decoration (otherwise `::` vs. `:` would be a problem already). So, with this approach you can split by 3 `___` to get the equivalent of `::` and by 2 `__` to get the equivalent of `@`, as long as you parse in that order. 

I could definitely use things like `___` vs. `---`, but I find them rather ugly :) WDYT?

Reply:
I like using `___` as the separator better, even though technically someone could create a key like `___` to mess with the merger, I don't think it's a real scenario. 

I don't know whether we want to just replace all keys in the spec, it's nice to use a `::` and `@` convention for readability/developer productivity.

Updated PR with `___`.